### PR TITLE
Fix case insensitive of ParserUtils.findDuplicates

### PR DESCRIPTION
**Pull request by jihoonson**
Dimension names should be case-sensitive.

### CODE REVIEW

**Comment by fjy** [comment]
Druid is case insensitive by design.

I'm not sure if there's a bug here. "Column" and "COLUMN" should be two separated columns in Druid.

**Comment by fjy** [comment]
@jihoonson can you fix test failure?

Also my last comment made no sense so I deleted it.

**Comment by drcrallen** [comment]
Is there any possible impact for sudden un-lower casing for people? This seems like one of those changes that seems simple but can break a lot of workflows in unexpected ways

**Comment by jihoonson** [comment]
@fjy thanks, fixed.

@drcrallen this PR is actually to allow same dimension names of different cases. For example, `dim` and `DIM` would be treated as different names. Since the dimension name was case-insensitive in some cases before, I think this PR wouldn't make any impact to existing workflows.

File: core/src/main/java/org/apache/druid/data/input/impl/DimensionsSpec.java [hunk 1]
```diff
@@ -199,9 +199,6 @@ private void verify(List<SpatialDimensionSchema> spatialDimensions)
         "dimensions and dimensions exclusions cannot overlap"
     );
 
-    ParserUtils.validateFields(dimNames);
```
Comment (clintropolis) [inline core/src/main/java/org/apache/druid/data/input/impl/DimensionsSpec.java]: it seems like this should still be there to check for true duplicates?

Comment (jihoonson) [reply core/src/main/java/org/apache/druid/data/input/impl/DimensionsSpec.java]: This check is duplicate. It's checked again here: https://github.com/apache/incubator-druid/pull/7692/files/20ad8911652e2625f15d0c93fd95fb4b48a4672e#diff-f7e4e56e91b3b0f304d4978038fe75c9R215.

Comment (clintropolis) [reply core/src/main/java/org/apache/druid/data/input/impl/DimensionsSpec.java]: Oh, i see, my bad :+1:

**Comment by jihoonson** [comment]
@fjy @drcrallen do you have more comments?

**Comment by jihoonson** [comment]
I'm merging this PR. @drcrallen Please let me know if you have more comments.

**Pull request merged**